app/scripts-browserify/search/es-aggregations-body.js

```diff
@@ -51,14 +51,158 @@ function generateDateRanges() {
       to: to.toString() + '||-1s'
     });
   }
-  // And beyound
+  // And beyond.
   result.push({
     from: lastYear.toString()
   });
   return result;
 }
 
-module.exports = function(parameters, body) {
+// Generate a series of filter-buckets for multi-field dateranges.
+function generateRangeRanges (fromField, toField) {
+  var result = {};
+
+  function getFilterEntry(fieldName, fieldValue, operator) {
+    let filter = {
+      'range': { }
+    };
+    filter.range[fieldName] =  {};
+    filter.range[fieldName][operator] = fieldValue;
+
+    // Should give us something like.
+    // {
+    //   'range': {
+    //     'creation_time_from.year': {
+    //       'gte': '1940'
+    //     }
+    //   }
+    // }
+    return filter;
+  }
+
+  function getEntry (fromValue, toValue) {
+    let entry = {
+      'bool': {
+        'must': []
+      }
+    };
+
+    if (fromValue) {
+      entry.bool.must.push(getFilterEntry(fromField, fromValue, 'gte'));
+    }
+
+    if (toValue) {
+      entry.bool.must.push(getFilterEntry(toField, toValue, 'lt'));
+    }
+
+    return entry;
+  }
+
+  // Start of with a openended range for <= CREATION_INTERVAL_FROM.
+  result['0'] = getEntry(false, CREATION_INTERVAL_FROM);
+
+  // Then setup a interval every 100 years up until 1900.
+  for(var y = CREATION_INTERVAL_FROM; y < 1900; y+= 100) {
+    let from = y;
+    let to = y + 100;
+
+    result[y] = getEntry(from, to);
+  }
+
+  // Then every ten years
+  var lastYear;
+  for(let y = 1900; y < CREATION_INTERVAL_TO; y+= 10) {
+    let from = y;
+    let to = y + 10;
+    lastYear = to;
+    result[from] = getEntry(from, to);
+  }
+
+  // Finish off with another openended range.
+  result[lastYear] = getEntry(lastYear, false);
+  return result;
+}
+
+/**
+ * Post-process a range-filter field.
+ */
+function postProcessRangeField (field, aggregationResult) {
+  // We've identified that field is of type date-range. Date-range filters can
+  // work with a single date-field (default) and a "multi" date fields
+  // (to/from)
+  // The Aggregation Result will have a
+  // aggregations.<fieldname>_independent.<fieldname> entry pr. single-field
+  // date-range field.
+  // If the filter is also setup with a multi-field range its result can be
+  // found under
+  // aggregations.<fieldname>_independent.<fieldname>_range
+  // We now go and find any multi-field results, and then augment the
+  // corresponding single-field results.
+
+  // Ensure we have all the data we need.
+  if (
+    !aggregationResult['aggregations']
+    || !aggregationResult['aggregations'][field + '_independent']
```

Comment:
A variable for `aggregationResult['aggregations'][field + '_independent']` would improve the readability.